var/spack/repos/builtin/packages/hadoop/package.py

```diff
@@ -13,9 +13,14 @@ class Hadoop(Package):
     """
 
     homepage = "http://hadoop.apache.org/"
-    url      = "http://apache-mirror.8birdsvideo.com/hadoop/common/hadoop-3.1.2/hadoop-3.1.2-src.tar.gz"
-
-    version('3.1.2', sha256='02f277c9fc75d4b13c7697ed6be30dba41667841ab5e37442855267c39bec6b6')
```

Comment:
Can you keep the 3.1.2 version?

Reply:
Infact there is no 3.1.2 version in the official mirror now and the old url is also broken, so I think this version should be abandoned?